src/ledger-bridge-keyring.ts

```diff
@@ -0,0 +1,883 @@
+  }
+
+  #getOrigin() {
+    const tmp = this.bridgeUrl.split('/');
+    tmp.splice(-1, 1);
+    return tmp.join('/');
+  }
+
+  #sendMessage(
+    msg: Partial<IFrameMessage>,
+    cb: (response: IFrameMessageResponse) => void,
+  ) {
+    msg.target = 'LEDGER-IFRAME';
+
+    this.currentMessageId += 1;
+    msg.messageId = this.currentMessageId;
+
+    this.messageCallbacks[this.currentMessageId] = cb;
+    this.iframe?.contentWindow?.postMessage(msg, '*');
+  }
+
+  #setupListener() {
+    this.#eventListener = ({ origin, data }) => {
+      if (origin !== this.#getOrigin()) {
+        return false;
+      }
+
+      if (data) {
+        if (this.messageCallbacks[data.messageId]) {
+          this.messageCallbacks[data.messageId]?.(data);
+        } else if (
+          data.action === IFrameMessageAction.LedgerConnectionChange &&
+          isConnectionChangedResponse(data.payload)
+        ) {
+          this.isDeviceConnected = data.payload.connected;
+        }
+      }
+
+      return undefined;
+    };
+    window.addEventListener('message', this.#eventListener);
+  }
+
+  destroy() {
+    if (this.#eventListener) {
+      window.removeEventListener('message', this.#eventListener);
+    }
+  }
+
+  async #getPage(increment: number) {
+    this.page += increment;
+
+    if (this.page <= 0) {
+      this.page = 1;
+    }
+    const from = (this.page - 1) * this.perPage;
+    const to = from + this.perPage;
+
+    await this.unlock();
+    let accounts;
+    if (this.#isLedgerLiveHdPath()) {
+      accounts = await this.#getAccountsBIP44(from, to);
+    } else {
+      accounts = this.#getAccountsLegacy(from, to);
+    }
+    return accounts;
+  }
+
+  async #getAccountsBIP44(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const path = this.#getPathForIndex(i);
+      const address = await this.unlock(path);
+      const valid = this.implementFullBIP44
+        ? await this.#hasPreviousTransactions(address)
+        : true;
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+
+      // PER BIP44
+      // "Software should prevent a creation of an account if
+      // a previous account does not have a transaction history
+      // (meaning none of its addresses have been used before)."
+      if (!valid) {
+        break;
+      }
+    }
+    return accounts;
+  }
+
+  #getAccountsLegacy(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const address = this.#addressFromIndex(pathBase, i);
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+      this.paths[ethUtil.toChecksumAddress(address)] = i;
+    }
+    return accounts;
+  }
+
+  // eslint-disable-next-line no-shadow
```

Comment:
I'm wondering if you can remove this now. I'm not sure what would be getting shadowed...